packages/cloud_firestore/cloud_firestore/lib/src/firestore.dart

```diff
@@ -8,154 +8,49 @@ part of cloud_firestore;
-    return DocumentReference._(this, path.split('/'));
-  }
+  DocumentReference document(String path) =>
+      DocumentReference._(_delegate.document(path));
 
-  /// Creates a write batch, used for performing multiple writes as a single
-  /// atomic operation.
-  ///
-  /// Unlike transactions, write batches are persisted offline and therefore are
-  /// preferable when you don’t need to condition your writes on read data.
-  WriteBatch batch() => WriteBatch._(this);
+  Future<void> enablePersistence(bool enable) =>
+      _delegate.enablePersistence(enable);
 
-  /// Executes the given TransactionHandler and then attempts to commit the
-  /// changes applied within an atomic transaction.
-  ///
-  /// In the TransactionHandler, a set of reads and writes can be performed
-  /// atomically using the Transaction object passed to the TransactionHandler.
-  /// After the TransactionHandler is run, Firestore will attempt to apply the
-  /// changes to the server. If any of the data read has been modified outside
-  /// of this transaction since being read, then the transaction will be
-  /// retried by executing the updateBlock again. If the transaction still
-  /// fails after 5 retries, then the transaction will fail.
-  ///
-  /// The TransactionHandler may be executed multiple times, it should be able
-  /// to handle multiple executions.
-  ///
-  /// Data accessed with the transaction will not reflect local changes that
-  /// have not been committed. For this reason, it is required that all
-  /// reads are performed before any writes. Transactions must be performed
-  /// while online. Otherwise, reads will fail, and the final commit will fail.
-  ///
-  /// By default transactions are limited to 5 seconds of execution time. This
-  /// timeout can be adjusted by setting the timeout parameter.
   Future<Map<String, dynamic>> runTransaction(
       TransactionHandler transactionHandler,
-      {Duration timeout = const Duration(seconds: 5)}) async {
-    assert(timeout.inMilliseconds > 0,
-        'Transaction timeout must be more than 0 milliseconds');
-    final int transactionId = _transactionHandlerId++;
-    _transactionHandlers[transactionId] = transactionHandler;
-    final Map<String, dynamic> result = await channel
-        .invokeMapMethod<String, dynamic>(
-            'Firestore#runTransaction', <String, dynamic>{
-      'app': app.name,
-      'transactionId': transactionId,
-      'transactionTimeout': timeout.inMilliseconds
-    });
-    return result ?? <String, dynamic>{};
-  }
-
-  @deprecated
-  Future<void> enablePersistence(bool enable) async {
-    assert(enable != null);
-    await channel
-        .invokeMethod<void>('Firestore#enablePersistence', <String, dynamic>{
-      'app': app.name,
-      'enable': enable,
-    });
+      {Duration timeout = const Duration(seconds: 5)}) {
+    return _delegate.runTransaction((platformTransaction) =>
+        transactionHandler(Transaction._(platformTransaction)));
   }
```

Comment:
I think you forgot to pass the `timeout` to the `_delegate.runTransaction` call:

```dart
return _delegate.runTransaction((platformTransaction) =>
        transactionHandler(Transaction._(platformTransaction)),
        timeout,
    );
```